docs/tutorials/preprocessing_tutorial.md

```diff
@@ -0,0 +1,174 @@
+
+## Preprocessing Layers
+Generally, we cannot feed the raw data into the NN layer. So, ElasticDL provides the preprocessing layers to transform the raw data into numeric tensors. 
+The preprocessing layers allow users to include data preprocessing directly in their Keras model. Now, the preprocessing layers mainly aim to transform
+structured data (numeric data or string data) stored as a CSV file or a MaxCompute table and so on.  
+
+### Transform numeric inputs
+For numeric inputs, ElasticDL provides `Normalizer` to scale the numeric data to a range, `Discretization` and other layers to map the numeric data to integer values.
+
+#### Normalizer Layer
+The `Normalizer` layer is to normalize the numeric values by (x-subtractor)/divisor. For example, we can set the subtractor to the minimum and divisor to the range size to 
+implement normalization.
+```python
+minimum = 3.0
+maximum = 7.0
+layer = Normalizer(subtractor=minimum, divisor=(maximum - minimum))
+input_data = tf.constant([[3.0], [5.0], [7.0]])
+result = layer(input_data)
+```
+If we want to implement standardization, we can set subtractor and divisor to the mean and standard deviation.
+
+#### Discretization layer
+The `Discretization` layer is to bucketize numeric data into discrete ranges according to boundaries and return integer values. For example, if the numeric
+data is `[19, 42, 55]` and the boundaries are `[30, 45]`, the outputs are `[0, 1, 2]`.
+
+```python
+age_values = tf.constant([[34, 45, 23, 67], [15, 37, 52, 47]])
+bins = [20, 30, 40, 50]
+layer = Discretization(bins=bins)
+result = layer(age_values)
+```
+The outputs are `[[2, 3, 1, 4], [0, 2, 4, 3]]`
+
+#### LogRound layer
+The `LogRound` layer is a special case of `Discretization` with fixed boundaries. It casts a numeric value into a discrete integer value by `round(log(x))`.
+The `base` of `LogRound` is the base of the `log` operator and the `num_bins` is the maximum output value. If the input value is bigger than `2^min_bins`, the output is also
+the `num_bins`. 
+```python
+    layer = LogRound(num_bins=16, base=2)
+    input_data = np.asarray([[1.2], [1.6], [0.2], [3.1], [100]])
+    result = layer(input_data)
+```
+The outputs are `[[0], [1], [0], [2], [7]]`
+
+
+#### RoundIdentity layer
+The `RoundIdentity` layer is to cast a float value to a integer value using `round(x)`. Then we can feed the integer values into `tf.keras.layer.Embedding`. If the input is bigger than
+the `max_value`, the output will be the `max_value`.
+ ```python
+    layer = RoundIdentity(max_value=5)
+    input_data = np.asarray([[1.2], [1.6], [0.2], [3.1], [4.9]])
+    result = layer(input_data)
+    
+```
+The outputs are `[[1], [2], [0], [3], [5]]`
+
+### Transform string inputs
+
+For string inputs, we usually make embedding of the inputs to numeric tensors. Using TensorFlow, We cannot feed the string inputs into `tf.keras.layer.Embedding` and must convert
```

Comment:
ElasticDL provides `Hashing` and `IndexLookup` (also add links) layers to map strings to numeric values. 

No need to mention embedding here, as we have another section later for embedding.